src/main/kotlin/assignment/Assignment.kt

```diff
@@ -20,14 +23,67 @@ fun main() {
     writeToFile(processedData)
 }
 
+
 fun readResponses(): List<Response> {
-    TODO()
-}
+    val inputStream: InputStream = File("src\\main\\kotlin\\assignment\\responses.txt").inputStream()
+    val responses = inputStream.bufferedReader().use { it.readLines() }
+    val studentresponses = mutableListOf<Response>()
+    val len1 = responses.lastIndex
+    for (i in 1..len1){
+        val len = responses[i].length
+        var j = 10
+        var n = 0
+        val branch = responses[i].slice(36..38)
+        val roll = responses[i].slice(22..28)
+        var covid = false
+        val cond = responses[i].slice(53..53)
```

Comment:
Try to figure out a solution where you don't have to depend on exact indices as sometimes student names can be very long and this indices approach won't work.